store/tikv/delete_range_test.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tikv
+
+import (
+	"bytes"
+	"context"
+	"math/rand"
+
+	. "github.com/pingcap/check"
+	"github.com/pingcap/tidb/store/mockstore/mocktikv"
+	"sort"
+)
+
+type testDeleteRangeSuite struct {
+	OneByOneSuite
+	cluster *mocktikv.Cluster
+	store   *tikvStore
+}
+
+var _ = Suite(&testDeleteRangeSuite{})
+
+func (s *testDeleteRangeSuite) SetUpTest(c *C) {
+	s.OneByOneSuite.SetUpSuite(c)
+	s.cluster = mocktikv.NewCluster()
+	mocktikv.BootstrapWithMultiRegions(s.cluster, []byte("a"), []byte("b"), []byte("c"))
+	client, pdClient, err := mocktikv.NewTestClient(s.cluster, nil, "")
+	c.Assert(err, IsNil)
+
+	store, err := NewTestTiKVStore(client, pdClient, nil, nil)
+	c.Check(err, IsNil)
+	s.store = store.(*tikvStore)
+}
+
+func (s *testDeleteRangeSuite) TearDownSuite(c *C) {
+	err := s.store.Close()
+	c.Assert(err, IsNil)
+	s.OneByOneSuite.TearDownSuite(c)
+}
+
+func (s *testDeleteRangeSuite) checkData(c *C, expectedData map[string]string) {
+	txn, err := s.store.Begin()
+	c.Assert(err, IsNil)
+	it, err := txn.Seek([]byte("a"))
+	c.Assert(err, IsNil)
+
+	// Scan all data and save into a map
+	data := map[string]string{}
+	for it.Valid() {
+		data[string(it.Key())] = string(it.Value())
+		err = it.Next()
+		c.Assert(err, IsNil)
+	}
+	txn.Commit(context.Background())
+
+	// Print log
+	var actualKeys []string
+	var expectedKeys []string
+	for key := range data {
+		actualKeys = append(actualKeys, key)
+	}
+	for key := range expectedData {
+		expectedKeys = append(expectedKeys, key)
+	}
+	sort.Strings(actualKeys)
+	sort.Strings(expectedKeys)
+	c.Log("Actual:   ", actualKeys)
```

Comment:
Is this debugging?

Reply:
Yes but it will only be printed out if some test failed. Should we remove it?